### DISCUSSION TITLE
Gestion des organisations 🏠

### DISCUSSION THREAD

**Discussion by agarrone** [datagouv/data.gouv.fr#1200] [Ideas]
Plusieurs équipes ont fait remonter le besoin d'avoir plus de finesse sur la gestion des droits au sein d'une organisation. Notamment pour la gestion des notifications ou la modération des commentaires. 
Cette discussion vise à permettre à chaque équipe de clarifier et spécifier son besoin.

**Comment by AntoineAugusti** [comment]
👋 Côté transport.data.gouv.fr, quelques notes sur ce sujet.

- Avoir la notion de service/d'équipes au sein d'une organisation. Une équipe regroupe plusieurs membres, avec des permissions différentes (créer un JDD, éditer un JDD, accéder à l'API, inviter des membres…
- Un JDD est sous la responsabilité/géré par une ou des équipes d'une organisation
- Envisager d'avoir des collaborateurs externes, pouvant uniquement réaliser certaines actions (par exemple répondre aux discussions)

J'aime beaucoup le modèle retenu par GitHub pour ce qui est des organisations, des équipes etc

**Comment by streino** [comment]
Mieux vaut tard que jamais... Synthèse des besoins AMDAC MTE et Ecosphères.

_J'utilise "item" pour généraliser dataset/topic/réutilisation/..., et "entité" pour généraliser orga/groupe/équipe/..._

### Métadonnées

Représenter les différents rôles qu'une organisation peut avoir avec un item : 

- Affecter un rôle différent de celui de "producteur". Par exemple l'IGN est diffuseur du RPG, pas producteur ; la DGALN fait produire des données par différents organismes qui se retrouvent dans différents catalogues (DREAL, Ifremer, ...).
- Capturer la notion de "pour le compte de"/"en délégation de" un autre rôle/orga. Par exemple "IGN diffuseur du RPG pour le compte de l'ASP producteur et du MASA MOA" est différent de "IGN diffuseur de la BDTOPAGE dans son Geoportail" (notion de légitimité dans les rôles).

Associer plusieurs organisations/rôles à un item : 

- Mapper l'ensemble des "rôles" identifiés dans les [métadonnées](https://github.com/ecolabdata/ecospheres/issues/43) des catalogues externes.
- Afficher plusieurs organisations sur la page d'un item + leurs rôles.
- Chercher/filtrer sur toutes les organisations d'un item + par rôle.
- Réexposer ces informations dans les expositions data.gouv (DCAT, API, CSV, ...).


### Gestion de patrimoine

Permettre aux administrateurs/responsables/coordinateurs/... de données de retrouver l'ensemble de leurs données (pourrait aussi aider les utilisateurs à naviguer data.gouv sans maîtriser l'organigramme du Service public ?) : 

- Cf besoins en section "Métadonnées" ci-dessus (multiples orgas + rôles).
- Modéliser les liens entre entités (orga-orga / orga-équipe) : 
  - Dans un premier temps, une hiérarchie à deux niveaux ("MTE > DG", "MTE > DDT", ...) pourrait déjà améliorer la situation actuelle mais le niveau DG/DDT est encore très macro. Au minimum on voudrait pouvoir descendre jusqu'à la sous-direction ("MTE > DGALN > DHUP > QV"). La demande DGALN est de pouvoir descendre le plus bas possible, mais se pose la question de la gestion pérenne aux plus bas niveaux.
  - Dans certains cas il peut être nécessaire de rattacher une organisation à plusieurs parents. Par exemple cas des partenariats (Hub'Eau : OFB et BRGM) ou des opérateurs avec plusieurs tutelles (IGN : MTECT et MASA).
- Déduire l'appartenance d'items à partir des liens entre organisations, sans qu'il soit nécessaire d'expliciter le lien item-orga au niveau de chaque item. Par exemple, rassembler les items provenant de DGs, DREALs, DDTs, etc. pour obtenir une vue d'ensemble du patrimoine MTE.
  Il serait intéressant de pouvoir inférer les liens entre organisations en reliant les orgas data.gouv aux entités administratives sur lannuaire.service-public.fr (en exploitant les métadonnées existantes dans les orgas data.gouv).
  - Pour les organismes publiques certifiés, relier les orgas data.gouv à leur(s) entité(s) officielles. Par exemple, l'orga data.gouv [Service central d’hydrométéorologie et d’appui à la prévision des inondations](https://www.data.gouv.fr/fr/organizations/service-central-dhydrometeorologie-et-dappui-a-la-prevision-des-inondations/) = la [page annuaire](https://lannuaire.service-public.fr/gouvernement/7e9f75fc-5862-4a33-8015-da82e7edfafd), ce qui permet d'inférer son appartenance au MTECT (note qu'il n'y a pas d'orga MTECT sur data.gouv mais [MTE](https://www.data.gouv.fr/fr/organizations/ministere-de-la-transition-ecologique/) et [MCT](https://www.data.gouv.fr/fr/organizations/ministere-de-l-egalite-des-territoires-et-du-logement/) car difficile de faire évoluer les orgas data.gouv ; ce problème pourrait être simplilfié en permettant d'affecter l'ancienne et la nouvelle orga à un jeu de données).
  - Pour les organisations data.gouv ne correspondant pas à des entités administratives mais portées par des organismes publics (groupes de travail, projets collaboratifs, applications, ...), on pourrait avoir une métadonnée différente qui identifie la/les entités lannuaire.service-public.fr dont dépend l'initiative (besoin d'indiquer le rôle et/ou les fondements juridiques sur lesquels le lien est établi ?).
- Définir un "périmètre de données" pouvant combiner hiérarchie et rôles. Par exemple, identifier l'ensemble des données sous la responsabilité de la DGALN (= orga DGALN + sous-entités hiérarchiques).


Faciliter l'administration des fiches de métadonnées : 

- Identifier les individus (et leurs orgas au moment de l'action) ayant publié/modifié une fiche, pour pouvoir gérer entre autres les réaffectation en cas d'évolution de structures.
- Identifier les items orphelins (items publiés par des individus ayant quitté l'orga responsable de publication).
- Réattribuer un moissonnage à une autre orga/équipe depuis l'admin pour gérer les cas d'évolution/restructuration de la hiérarchie.


### Bouquets

_Besoins déjà largement identifiés pour d'autres types d'items._

Permettre la collaboration de plusieurs entités autour d'un bouquet : 

- Attribuer différents rôles à différentes entités.
- Attribuer des droits au niveau organisation ou individu.
- Attribuer des droits à des sous-entités précises (moindre privilège).
- Hériter les droits d'organisation(s) parente(s) + spécialiser les droits nécessaires, pour faciliter l'administration des droits.
- Dans le cas de bouquets privés/restreints, restreindre l'accès à une liste de personnes/orgas explicitement définie (whitelist).

Rôles envisagés : idem jeux de donnés (producteur, diffuseur, commanditaire, ...).

Droits envisagés : 
- Super-admin : Peut tout faire sur tous les bouquets = admin ecologie.data.gouv (sans pour autant être admin data.gouv ?).
- Admin : Peut tout faire sur les bouquets de son/ses orgas.
- Editeur / créateur : Peut créer/modifier des bouquets au sein de son/ses orgas (regrouper avec Admin ?).
- Utilisateur : Peut commenter un bouquet, peut suggérer des modifications aux bouquets qui devront être validées par un Editeur/Admin.

Exemple bouquet "droit du sol" : 
- Bouquet sur un territoire donné : 
  - Admins : DDT du territoire.
  - Editeurs : Géomaticien DDT, instructeur du droit des sols, bureau d'étude, développeur accrédité d'une application aval, ...
- Trame nationale : 
  - Admins : Service en AC ou DDT porteuse.
  - Editeurs : Toutes les DDT.

**Reply by agarrone** [reply]
Merci 🙏
